### PR TITLE
Fix: site studio already present

### PR DESCRIPTION
**Pull request by kalisp**
## Changelog Description
Added protection from duplicate sites.

## Additional review information
Multiple clients reported issue in `integrate_site_sync` with `Site already present`, couldnt reproduce on our side, but this protection shouldn't hurt

### Error
```
"/ayon/addons/sitesync_1.2.0/ayon_sitesync/addon.py", line 220, in add_site 
    raise SiteAlreadyPresentError(msg) 
ayon_sitesync.utils.SiteAlreadyPresentError: Site studio already present 
!!! ERR: 
{ CLI-publish }: [ Failed IntegrateSiteSync: Site studio already present -- ('/ayon/addons/sitesync_1.2.0/ayon_sitesync/plugins/publish/integrate_site_sync.py', 220, 'add_site', 'raise SiteAlreadyPresentError(msg)') ]
```
Issue linked https://app.clickup.com/t/6658547/AY-7096
Should close https://github.com/ynput/ayon-sitesync/issues/42

## Testing notes:
1. publish something with SiteSync enabled

### CODE REVIEW

File: client/ayon_sitesync/addon.py [hunk 1]
```diff
@@ -338,24 +338,30 @@ def create_metadata(name, created=True):
...
-
-        return list(attached_sites.values())
+        for site_name in self._get_always_accessible_sites(project_name):
+            if site_name not in attached_sites:
+                attached_sites[site_name] = (
+                    create_metadata(site_name, created=False))
+        unique_sites = {}
+        for site in attached_sites.values():
+            unique_sites[site["name"]] = site
+        return list(unique_sites.values())
```
Comment (BigRoy) [inline client/ayon_sitesync/addon.py:348]: I don't get it? it's already unique in the dict by name. The `name` key in the result of `create_metadata` is exactly that `site_name`?

**Comment by kalisp** [comment]
I couldnt reproduce mentioned issue, but two clients reported it. I think representation_id must be unique https://github.com/ynput/ayon-sitesync/blob/develop/client/ayon_sitesync/plugins/publish/integrate_site_sync.py#L31, so only place I though could get site duplicated was here.

**Comment by iLLiCiTiT** [comment]
> I couldnt reproduce mentioned issue, but two clients reported it. I think representation_id must be unique https://github.com/ynput/ayon-sitesync/blob/develop/client/ayon_sitesync/plugins/publish/integrate_site_sync.py#L31, so only place I though could get site duplicated was here.

There is no information what this does fix, not link to issue neither traceback.

**Comment by iLLiCiTiT** [comment]
Maybe a question is why does this raises an error? I mean, why it is failure? You wanted to add representation to site, it is already there, it should just skip the rest of logic, or?
```python
    def add_site(...):
        ...
        if not force:
            existing = self.get_repre_sync_state(
                project_name,
                representation_id,
                site_name
            )
            if existing:
                failure = True
                if file_id:
                    file_exists = existing.get("files", {}).get(file_id)
                    if not file_exists:
                        failure = False

                if failure:
                    msg = "Site {} already present".format(site_name)
                    self.log.info(msg)
                    raise SiteAlreadyPresentError(msg)
```

**Comment by kalisp** [comment]
I am checking for existing only if not Force. If force, current state doesn't matter, purge all, start fresh. If not Force it checks if site exists already, it could exist for representations, but we might be missing/adding specific `file_id`.

**Review by MilaKudr — APPROVED** [review]
Working for me, no errors. Syncing all files...

File: client/ayon_sitesync/addon.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (iLLiCiTiT) [reply client/ayon_sitesync/addon.py:348]: I have some questions... Like how do we know it uses correct site information? Or are those with same name having the same information?

[the remark this answers is not in the document]
Comment (kalisp) [reply client/ayon_sitesync/addon.py]: They have same values. Date created or state that they should be synced.

**Pull request merged**